src/main/java/com/puppycrawl/tools/checkstyle/checks/naming/ConstantNameCheck.java

```diff
@@ -153,12 +153,11 @@ protected final boolean mustCheckName(DetailAST ast) {
 
         final DetailAST modifiersAST =
             ast.findFirstToken(TokenTypes.MODIFIERS);
-        final boolean isStatic = modifiersAST.findFirstToken(TokenTypes.LITERAL_STATIC) != null;
-        final boolean isFinal = modifiersAST.findFirstToken(TokenTypes.FINAL) != null;
-
-        if (isStatic && isFinal && shouldCheckInScope(modifiersAST)
-                || ScopeUtil.isInAnnotationBlock(ast)
-                || ScopeUtil.isInInterfaceBlock(ast)
+        final boolean isStaticFinal = modifiersAST.findFirstToken(TokenTypes.LITERAL_STATIC) != null
+            && modifiersAST.findFirstToken(TokenTypes.FINAL) != null
+            || ScopeUtil.isInAnnotationBlock(ast)
+            || ScopeUtil.isInInterfaceBlock(ast);
```

Comment:
```
        final boolean isStaticFinal = 
            modifiersAST.findFirstToken(TokenTypes.LITERAL_STATIC) != null
                && modifiersAST.findFirstToken(TokenTypes.FINAL) != null
            || ScopeUtil.isInAnnotationBlock(ast)
            || ScopeUtil.isInInterfaceBlock(ast);
```

to make it more visual priority of calculation

Reply:
done